packages/paste-core/components/heading/src/types.ts

```diff
@@ -1,11 +1,11 @@
 import type {TextProps} from '@twilio-paste/text';
+import type {HTMLPasteProps} from '@twilio-paste/types';
 
 export type AsTags = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'div' | 'label' | 'span' | 'header';
 export type HeadingVariants = 'heading10' | 'heading20' | 'heading30' | 'heading40' | 'heading50' | 'heading60';
 
-export interface HeadingProps extends React.HTMLAttributes<HTMLHeadingElement>, Pick<TextProps, 'element' | 'display'> {
+export interface HeadingProps extends HTMLPasteProps<AsTags>, Pick<TextProps, 'element' | 'display'> {
```

Comment:
question: why is this one AsTags

Reply:
Because it can be one of either `'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'div' | 'label' | 'span' | 'header'`